test/core/end2end/tests/grpc_authz.cc

```diff
@@ -546,6 +540,16 @@ static void test_file_watcher_valid_policy_reload(
           tmp_policy.name().c_str(), /*refresh_interval_sec=*/1, &code,
           &error_details);
   GPR_ASSERT(GRPC_STATUS_OK == code);
+  absl::Notification on_reload_done;
+  std::function<void(grpc_status_code code, const char* error_details)>
+      callback = [&on_reload_done](grpc_status_code status,
+                                   const char* error_details) {
+        GPR_ASSERT(GRPC_STATUS_OK == status);
+        GPR_ASSERT(nullptr == error_details);
+        on_reload_done.Notify();
+      };
+  dynamic_cast<grpc_core::FileWatcherAuthorizationPolicyProvider*>(provider)
+      ->SetCallbackForTesting(std::move(callback));
```

Comment:
To be safe, I think we should reset the callback to null at the end of each test, to make sure that the lambda does not get called again after we return from this function, because the state captured by the lambda will no longer be valid then.

Same thing in all tests.